components/script/dom/htmltitleelement.rs

```diff
@@ -69,17 +69,17 @@ impl VirtualMethods for HTMLTitleElement {
             s.children_changed(mutation);
         }
         let node = self.upcast::<Node>();
-        if node.is_in_doc() {
+        if node.is_connected() {
             node.owner_doc().title_changed();
```

Comment:
This `is_connected` change is wrong. You don't want a `<title>` in a shadow root to affect the document.